executor/src/constant_evaluator/mod.rs

```diff
@@ -49,8 +49,13 @@ fn generate_values<T: FieldElement>(
                 // We could try to avoid the first evaluation to be run for each iteration,
                 // but the data is not thread-safe.
                 let fun = evaluator::evaluate(e, &symbols).unwrap();
-                evaluator::evaluate_function_call(fun, vec![Rc::new(T::from(i).into())], &symbols)
-                    .and_then(|v| v.try_to_number())
+                evaluator::evaluate_function_call(
+                    fun,
+                    vec![Rc::new(Value::Integer(num_bigint::BigInt::from(i)))],
+                    &symbols,
+                )
+                // TODO should this be non-sloppy?
```

Comment:
I think it's OK to make this non-sloppy. This means that if we have a function defining a fixed column that has e.g. negative numbers, we have to explicitly call a function to reduce it into the field.

Reply:
makes sense